CPAC/resources/configs/pipeline_config_ccs-options.yml

```diff
@@ -0,0 +1,279 @@
+
+FROM: default
+
+
+pipeline_setup: 
+
+  # Name for this pipeline configuration - useful for identification.
+  pipeline_name: cpac_ccs-options
+
+  output_directory: 
+
+    # Directory where C-PAC should write out processed data, logs, and crash reports.
+    # - If running in a container (Singularity/Docker), you can simply set this to an arbitrary
+    #   name like '/output', and then map (-B/-v) your desired output directory to that label.
+    # - If running outside a container, this should be a full path to a directory.
+    path: /outputs/output
+
+    # Include extra versions and intermediate steps of functional preprocessing in the output directory.
+    write_func_outputs: True
+
+  working_directory: 
+
+    # Directory where C-PAC should store temporary and intermediate files.
+    # - This directory must be saved if you wish to re-run your pipeline from where you left off (if not completed).
+    # - NOTE: As it stores all intermediate files, this directory can grow to become very
+    #   large, especially for data with a large amount of TRs.
+    # - If running in a container (Singularity/Docker), you can simply set this to an arbitrary
+    #   name like '/work', and then map (-B/-v) your desired output directory to that label.
+    # - If running outside a container, this should be a full path to a directory.
+    # - This can be written to '/tmp' if you do not intend to save your working directory.
+    path: /outputs/working
+
+  log_directory: 
+
+    path: /outputs/log
+
+  system_config: 
+
+    # The maximum amount of memory each participant's workflow can allocate.
+    # Use this to place an upper bound of memory usage.
+    # - Warning: 'Memory Per Participant' multiplied by 'Number of Participants to Run Simultaneously'
+    #   must not be more than the total amount of RAM.
+    # - Conversely, using too little RAM can impede the speed of a pipeline run.
+    # - It is recommended that you set this to a value that when multiplied by
+    #   'Number of Participants to Run Simultaneously' is as much RAM you can safely allocate.
+    maximum_memory_per_participant: 10.0
+
+# PREPROCESSING
+# -------------
+surface_analysis: 
+
+  # Will run Freesurfer for surface-based analysis. Will output traditional Freesurfer derivatives.
+  # If you wish to employ Freesurfer outputs for brain masking or tissue segmentation in the voxel-based pipeline,
+  # select those 'Freesurfer-' labeled options further below in anatomical_preproc.
+  run_freesurfer: On
+
+  # Add extra arguments to recon-all command
+  reconall_args: '-clean-bm -gcut'
+
+anatomical_preproc: 
+
+  # Non-local means filtering via ANTs DenoiseImage
+  non_local_means_filtering: [On]
+
+  brain_extraction:
+
+    # using: ['3dSkullStrip', 'BET', 'UNet', 'niworkflows-ants', 'FreeSurfer-ABCD', 'FreeSurfer-BET-Tight', 'FreeSurfer-BET-Loose']
+    # this is a fork option
+    using: ['FreeSurfer-BET-Tight', 'FreeSurfer-BET-Loose']
```

Comment:
Currently, we'll generate 3 anatomical masks in ccs-options config: default FreeSurfer mask, FreeSurfer-BET tight mask and FreeSurfer-BET loose mask. Even though we generate 3 masks, it'll only use default FreeSurfer mask to do brain extraction if no manual intervention. CCS will generate 3 masks at the first stage and let users choose the best mask for following steps. I am wondering if there's a better way to replicate CCS. Do we need to fork 3 pipelines with 3 anatomical masks?

Reply:
Like we've decided in the past for other "manual checkpoint" use cases - we can offer both.

- Users can simply enable only the mask generation and run the pipeline, then pick which mask they want and run the rest of the pipeline.
- Or, can fork all 3 and let CPAC blast through all options.